### PR TITLE
[Improvement][API-Test] Fix api test dependencies

### PR DESCRIPTION
**Pull request by EricGao888**
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Review by ruanwenjun — DISMISSED** [review]
LGTM, wait CI pass.

**Comment by ruanwenjun** [comment]
`dolphinscheduler-api-test` should  put into `module`

**Comment by EricGao888** [comment]
image

**Comment by EricGao888** [comment]
> `dolphinscheduler-api-test` should put into `module`

@ruanwenjun  Will this help wipe out the errors?

It seems the dependencies installed in `build` job cannot be accessed during `api-test` job.

image

**Comment by EricGao888** [comment]
I'm trying to force `build` job and `api-test` job to run on the same machine.

**Comment by kezhenxu94** [comment]
The cache step should already fetch back the dependencies that were installed in previous steps. Rebuilding the modules in tests can fix the issue but waste more time

File: .github/workflows/api-test.yml [hunk 1]
```diff
@@ -44,9 +52,9 @@ jobs:
               - '!(docs/**)'
   build:
     name: API-Test-Build
-    needs: paths-filter
+    needs: [ paths-filter, get-runner ]
     if: ${{ (needs.paths-filter.outputs.not-ignore == 'true') || (github.event_name == 'push') }}
-    runs-on: ubuntu-latest
+    runs-on: ${{needs.get-runner.outputs.RUNNER}}
```
Comment (kezhenxu94) [inline .github/workflows/api-test.yml]: There is no guarantee that the runner that runs previous steps can be picked up again in this step.

**Comment by ruanwenjun** [comment]
> Will this help wipe out the errors

Yes, you can try.

**Comment by SbloodyS** [comment]
May be you can try to replace all `actions/checkout@v2` to `actions/checkout@v3` in https://github.com/apache/dolphinscheduler/blob/dev/.github/workflows/api-test.yml

@EricGao888

**Comment by kezhenxu94** [comment]
> May be you can try to replace all `actions/checkout@v2` to `actions/checkout@v3` in https://github.com/apache/dolphinscheduler/blob/dev/.github/workflows/api-test.yml
> 
> @EricGao888

The CI failed because of maven dependencies, `actions/checkout` is all about source code, why are they related?

**Comment by SbloodyS** [comment]
> > May be you can try to replace all `actions/checkout@v2` to `actions/checkout@v3` in https://github.com/apache/dolphinscheduler/blob/dev/.github/workflows/api-test.yml
> > @EricGao888
> 
> The CI failed because of maven dependencies, `actions/checkout` is all about source code, why are they related?

Although I think it's okay, giving it a try is always good~

**Comment by SbloodyS** [comment]
Related PR: #14808

**Comment by ruanwenjun** [comment]
Closed by #14808

**Comment by EricGao888** [comment]
Actually I was planning to fix style issues in this PR as well. Since it is closed, I will open another one to fix style issues in api-test and e2e module...

**Pull request closed without merging**